frappe/public/js/frappe/form/grid.js

```diff
@@ -502,10 +502,9 @@ export default class Grid {
 
 	set_column_disp(fieldname, show) {
 		if ($.isArray(fieldname)) {
-			for (var i = 0, l = fieldname.length; i < l; i++) {
-				var fname = fieldname[i];
-				this.get_docfield(fname).hidden = show ? 0 : 1;
-				this.set_editable_grid_column_disp(fname, show);
+			for (let field of fieldname) {
+				this.update_docfield_property(field, "hidden", show);
```

Comment:
@hrwX probably a typo in this line. By setting hidden to the value of "show" we hide the fields when show is true, thus doing the opposite of what's expected.

Reply:
Fixed in https://github.com/frappe/frappe/pull/16255

Reply:
Already solved in https://github.com/frappe/frappe/pull/16255